File: php/features.php
<?php
declare(strict_types=1);

// PublibikeStations SDK feature factory

require_once __DIR__ . '/feature/BaseFeature.php';
require_once __DIR__ . '/feature/TestFeature.php';


class PublibikeStationsFeatures
{
    public static function make_feature(string $name)
    {
        switch ($name) {
            case "base":
                return new PublibikeStationsBaseFeature();
            case "test":
                return new PublibikeStationsTestFeature();
            default:
                return new PublibikeStationsBaseFeature();
        }
    }
}

File: py/publibikestations_sdk/features.py
# PublibikeStations SDK feature factory

from publibikestations_sdk.feature.base_feature import PublibikeStationsBaseFeature
from publibikestations_sdk.feature.test_feature import PublibikeStationsTestFeature


def _make_feature(name):
    features = {
        "base": lambda: PublibikeStationsBaseFeature(),
        "test": lambda: PublibikeStationsTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
